src/atomicassets.cpp

```diff
@@ -1221,6 +1285,96 @@ ACTION atomicassets::logburnasset(
 }
 
 
+/**
+* Function for creating a template, handling both the creation of normal templates & purely mutable templates
+*/
+
+void atomicassets::create_template(
+    name & authorized_creator,
+    name & collection_name,
+    name & schema_name,
+    bool transferable,
+    bool burnable,
+    uint32_t max_supply,
+    ATTRIBUTE_MAP & immutable_data,
+    ATTRIBUTE_MAP mutable_data
+) { 
+    require_auth(authorized_creator);
+
+    auto collection_itr = collections.require_find(collection_name.value,
+        "No collection with this name exists");
+
+    check_has_collection_auth(
+        authorized_creator,
+        collection_name,
+        "The creator is not authorized within the collection"
+    );
+
+    schemas_t collection_schemas = get_schemas(collection_name);
+    auto schema_itr = collection_schemas.require_find(schema_name.value,
+        "No schema with this name exists");
+
+    config_s current_config = config.get();
+    int32_t template_id = current_config.template_counter++;
+    config.set(current_config, get_self());
+
+    if (transferable == false){
```

Comment:
I'd prefer
`check(burnable || transferable, "A template cannot be both non-transferable and non-burnable")`

Reply:
clean